server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -991,6 +1022,10 @@ public abstract VectorData parseKnnVector(
 
         public abstract int getNumBytes(int dimensions);
 
+        public abstract int getEmbeddingLength(int dimensions);
+
+        public abstract int getDimensions(int embeddingLength);
```

Comment:
Are the two new methods only required for testing? It's a bit unclear. Would it be possible to generate the embeddings in tests based on the element type instead? The dims parameter of the dense vector field should be sufficient to generate random vectors according to their element type.